contracts/FundManagement.sol

```diff
@@ -0,0 +1,131 @@
+            "No withdrawable funds"
+        );
+        require(
+            fund.amountWithdrawable <= fund.totalAmount, 
+            "Withdrawable amount greater than total allocated funds"
+        );
+        require(
+            block.timestamp >= fund.requestWithdrawTime + requestWindow, 
+            "Your funds are not withdrawable yet"
+        );
+
+        require(
+            IERC20(fund.asset).transfer(msg.sender, fund.amountWithdrawable), 
+            "Transfer failed"
+        );
+
+        emit Claim(msg.sender, fund.amountWithdrawable);
+
+        locked[fund.asset] = locked[fund.asset] - fund.amountWithdrawable;
+        fund.totalAmount = fund.totalAmount - fund.amountWithdrawable;
+        fund.amountWithdrawable = 0;
+    }
+
+    function setRequestWindow(uint256 duration) external onlyOwner {
+        requestWindow = duration;
+    }
+
+    function createFund(address account, uint256 amount, address asset) external onlyOwner {
+        require(
+            amount > 0,
+            "Invalid amount"
+        );
+        uint256 currentLocked = locked[asset];
+        require(
+            IERC20(asset).balanceOf(address(this)) >= currentLocked + amount,
+            "Not enough tokens"
+        );
+
+        uint256 currentNumFunds = numberOfFunds[account];
+        funds[account][currentNumFunds] = Fund(
+            amount,
+            asset,
+            0,
+            0
+        );
+
+        numberOfFunds[account] = currentNumFunds + 1;
+        locked[asset] = currentLocked + amount;
+
+        emit CreateFund(account, asset, amount);
+    }
+
+    function clawbackFunds(address account, uint256 fundNumber) external onlyOwner {
+        Fund storage fund = funds[account][fundNumber];
+
+        locked[fund.asset] = locked[fund.asset] - fund.totalAmount;
+        fund.totalAmount = 0;
+        fund.amountWithdrawable = 0;
+
+        emit Clawback(account, fundNumber);
+    }
+
+    function addToFund(address account, uint256 amount, uint256 fundNumber) external onlyOwner {
+        Fund storage fund = funds[account][fundNumber];
+        uint256 currentLocked = locked[fund.asset];
+        require(
+            IERC20(fund.asset).balanceOf(address(this)) - currentLocked >= amount,
+            "Not enough unlocked tokens to add"
+        );
+
+        fund.totalAmount = fund.totalAmount + amount;
+        locked[fund.asset] = currentLocked + amount;
+    }
+
+    function withdrawUnlockedAssets(uint256 amount, address asset) external onlyOwner {
+        IERC20 token = IERC20(asset);
+        require(
+            token.balanceOf(address(this)) - locked[asset] >= amount,
+            "Not enough unlocked tokens to withdraw"
+        );
+        require(token.transfer(owner(), amount), "Withdraw failed");
+    }
+
+    function emergencyWithdraw(uint256 amount, address asset) external onlyOwner {
+        IERC20 token = IERC20(asset);
+        require(token.transfer(owner(), amount), "Withdraw failed");
```

Comment:
Might be worth not having this, or ensuring the asset isn't an asset that is locked or something? Otherwise the `withdrawUnlockedAssets` function is kind of pointless.

Reply:
clawback() and withdrawUnlockedAssets() should cover all the withdrawal needs i wrote this incase we have many funds under the owner and needs to do a withdrawal quickly rather than clawback every fund first. Happy to remove it if that doesn't seem like something we need.